fix crash on x % 0`

tbg opened the pull request:
found via go-fuzz

ping @dvyukov for your trophy list.

dvyukov commented:
added to trophy list

tamird commented on sql/parser/eval.go:
```diff
@@ -231,6 +232,10 @@ var binOps = map[binArgs]func(Datum, Datum) (Datum, error){
 	},
 
 	binArgs{Mod, intType, intType}: func(left Datum, right Datum) (Datum, error) {
+		r := right.(DInt)
```
nit: no need for this local

tschottdorf replied:
whoops, meant to use it below instead of `right.(DInt)`. Will fix.

tamird replied:
:+1:

tamird commented:
LGTM

tbg committed:
fix crash on x % 0`

1 % 0 now returns an error (as does Postgres 9.3; MySQL goes for NULL)

found via go-fuzz

petermattis commented:
LGTM

See #1622.
